### PR TITLE
Added sort to load mail templates from selected theme first, if exists

### PR DESCRIPTION
**Pull request by memoworker**

### CODE REVIEW

File: src/Services/MailFinderService.php [hunk 1]
```diff
@@ -67,6 +82,25 @@ public function findTemplateByTechnicalName(
 
         $searchFolder = $this->searchPathProvider->buildPaths($businessEvent);
 
+        $criteria = (new Criteria)->addFilter(new EqualsFilter('salesChannels.id', $businessEvent->getSalesChannelId()));
+        $searchResult = $this->themeRepository->search($criteria, $businessEvent->getContext());
```
Comment (shyim) [inline src/Services/MailFinderService.php:86]: this fetch is useless, you overwrite it few lines below

File: src/Services/MailFinderService.php [hunk 2]
```diff
@@ -67,6 +82,25 @@ public function findTemplateByTechnicalName(
...
+
+        $criteria = (new Criteria)->addFilter(new EqualsFilter('name', $searchResult->first()->getTechnicalName()));
```
Comment (shyim) [inline src/Services/MailFinderService.php:88]: Can you do this with plain sql?

Also an Theme can be an app

Comment (memoworker) [reply src/Services/MailFinderService.php]: @shyim we changed this to be plain sql and added support for the new feature from pull request #35 which we have not tested before.
Sadly it is not possible to have the correct path displayed in the Shopware Backend with our change, as this change makes the actual path related to the selected sales channel and theme (if the theme contains mail templates)

File: src/Services/MailFinderService.php [hunk 3]
```diff
@@ -82,24 +76,28 @@ public function findTemplateByTechnicalName(
...
         $searchFolder = $this->searchPathProvider->buildPaths($businessEvent);
 
-        $criteria = (new Criteria)->addFilter(new EqualsFilter('salesChannels.id', $businessEvent->getSalesChannelId()));
-        $searchResult = $this->themeRepository->search($criteria, $businessEvent->getContext());
+        $themePath = $this->connection->fetchOne(
+            'SELECT IFNULL(a.path, p.path) AS `path` FROM `theme` AS t '
+                    . 'LEFT JOIN `theme_sales_channel` AS tsc ON tsc.`theme_id` = t.`id` '
+                    . 'LEFT JOIN `plugin` AS p ON p.`name` = t.`technical_name` '
+                    . 'LEFT JOIN `app` AS a ON a.`name` = t.`technical_name` '
+                    . 'WHERE tsc.`sales_channel_id` = UNHEX("' . $businessEvent->getSalesChannelId() . '");'
```
Comment (shyim) [inline src/Services/MailFinderService.php:83]: Can you use prepared statements? Uuid::fromHexToBytes helps you also to convert it

**Pull request merged**